internal/services/nginx/nginx_configuration_resource.go

```diff
@@ -0,0 +1,318 @@
+		}
+	}
+
+	return req
+}
+
+type ConfigurationResource struct{}
+
+var _ sdk.Resource = (*ConfigurationResource)(nil)
+
+func (m ConfigurationResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"nginx_deployment_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"config_file": {
+			Type:     pluginsdk.TypeList,
+			Required: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"content": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsBase64,
+					},
+
+					"virtual_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"protected_file": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"content": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsBase64,
+					},
+
+					"virtual_path": {
+						Type:         pluginsdk.TypeString,
+						Required:     true,
+						ValidateFunc: validation.StringIsNotEmpty,
+					},
+				},
+			},
+		},
+
+		"package_data": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"root_file": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+	}
+}
+
+func (m ConfigurationResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		// name of nginx configuration set to a fix value `default` by service team.
+		"name": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+}
+
+func (m ConfigurationResource) ModelObject() interface{} {
+	return &ConfigurationModel{}
+}
+
+func (m ConfigurationResource) ResourceType() string {
+	return "azurerm_nginx_configuration"
+}
+
+func (m ConfigurationResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			client := meta.Client.Nginx.NginxConfiguration
+
+			var model ConfigurationModel
+			if err := meta.Decode(&model); err != nil {
+				return err
+			}
+
+			model.Name = defaultConfigurationName
+			deployID, err := nginxdeployment.ParseNginxDeploymentID(model.NginxDeploymentId)
+			if err != nil {
+				return err
+			}
+
+			subscriptionID := meta.Client.Account.SubscriptionId
+			id := nginxconfiguration.NewConfigurationID(subscriptionID, deployID.ResourceGroupName, deployID.DeploymentName, model.Name)
+			// get/list, get will cause internal server error if default configuration not exists
+			// todo remove set retry to 1
+			client.Client.RetryAttempts = 1
```

Comment:
is there a Swagger issue tracking that?

Reply:
it looks like the server working fine now. removed these lines.